src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -359,147 +360,180 @@ async Task<JObject> ResolveAsInstanceMember(ArraySegment<string> parts, JObject
+                        var typeIds = await context.SdbAgent.GetTypeIdsForObject(objectId.Value, true, token);
+                        int[] methodIds = await context.SdbAgent.GetMethodIdsByName(typeIds[0], "get_Item", BindingFlags.Default, token);
+                        if (methodIds == null)
+                            throw new InvalidOperationException($"Type '{rootObject?["className"]?.Value<string>()}' cannot be indexed.");
+
+                        // get_Item should not have an overload, but if user defined it, take the default one: with one param (key)
+                        int allowedParamCnt = 1;
+                        int toArrayId = methodIds[0];
+                        // ToDo: optimize the loop by choosing the right method at once without trying out them all
+                        for (int i = 0; i < methodIds.Length; i++)
+                        {
+                            MethodInfoWithDebugInformation methodInfo = await context.SdbAgent.GetMethodInfo(methodIds[i], token);
+                            ParameterInfo[] paramInfo = methodInfo.GetParametersInfo();
+
+                            if (paramInfo.Length == allowedParamCnt)
                             {
-                                var arg = elementAccess.ArgumentList.Arguments[i];
-                                if (i != 0)
+                                try
                                 {
-                                    elementIdxStr += ", ";
-                                    multiDimensionalArray = true;
+                                    if (indexObject is null && elementIdxInfo.IndexingExpression is null)
+                                        throw new InternalErrorException($"Unable to write index parameter to invoke the method in the runtime.");
+                                    ArraySegment<byte> buffer = indexObject is null ?
+                                        await WriteLiteralExpressionAsIndex(objectId,  elementIdxInfo.IndexingExpression, elementIdxInfo.ElementIdxStr) :
+                                        await WriteJObjectAsIndex(objectId, indexObject, elementIdxInfo.ElementIdxStr);
+                                    JObject getItemRetObj = await context.SdbAgent.InvokeMethod(buffer, methodIds[i], token);
+                                    return (JObject)getItemRetObj["value"];
                                 }
-                                // e.g. x[1]
-                                if (arg.Expression is LiteralExpressionSyntax)
+                                catch (Exception ex)
                                 {
-                                    var argParm = arg.Expression as LiteralExpressionSyntax;
-                                    elementIdxStr += argParm.ToString();
+                                    logger.LogDebug($"Attempt number {i + 1} out of {methodIds.Length} of invoking method {methodInfo.Name} failed. Method Id = {methodIds[i]}.\nInner exception: {ex}.");
```

Comment:
A more useful message would be to print the container type, and the parameter type.

Also, in a follow up, we should check the for a compatible parameter type.

Reply:
`ParameterInfo` generally does not have an info about the type unless the parameter is a default one. It's a really rare situation. We can show the param's name instead, maybe it will be helpful (but not in all situations). About the container type - it's a good idea.